ratatui/src/widgets.rs

```diff
@@ -1,16 +1,513 @@
-//! crate.
+//! They are used to create and manage the layout and style of the terminal interface. Widgets can
+//! be combined and nested to create complex UIs, and can be easily customized to suit the needs of
+//! your application.
 //!
-//! Widgets are created for each frame as they are consumed after rendered. They are not meant to be
-//! stored but used as *commands* to draw common figures in the UI.
+//! Ratatui provides a wide variety of built-in widgets that can be used to quickly create UIs.
+//! Additionally, [`String`], [`&str`], [`Span`], [`Line`], and [`Text`] can be used as widgets
+//! (though often [`Paragraph`] is used instead of these directly as it allows wrapping and
+//! surrounding the text with a block).
+//!
+//! # Crate Organization
+//!
+//! Starting with Ratatui 0.30.0, the project was split into multiple crates for better modularity:
+//!
+//! - **[`ratatui-core`]**: Contains the core widget traits ([`Widget`], [`StatefulWidget`]) and
+//!   text-related types ([`String`], [`&str`], [`Span`], [`Line`], [`Text`])
+//! - **[`ratatui-widgets`]**: Contains all the built-in widget implementations ([`Block`],
+//!   [`Paragraph`], [`List`], etc.)
+//! - **[`ratatui`](crate)**: The main crate that re-exports everything for convenience. The
+//!   unstable [`WidgetRef`] and [`StatefulWidgetRef`] traits are defined in the main `ratatui`
+//!   crate as they are experimental.
+//!
+//! This split serves different user needs:
+//!
+//! - **App Authors**: Most application developers should use the main [`ratatui`](crate) crate,
+//!   which provides everything needed to build terminal applications with widgets, backends, and
+//!   layout systems
+//! - **Widget Library Authors**: When creating third-party widget libraries, consider depending
+//!   only on [`ratatui-core`] to avoid pulling in unnecessary built-in widgets and reduce
+//!   compilation time for your users
+//! - **Minimalist Projects**: Use [`ratatui-core`] directly if you only need the fundamental traits
+//!   and text types without any built-in widgets
+//!
+//! The modular structure allows widget library authors to create lightweight dependencies while
+//! still being compatible with the broader Ratatui ecosystem.
+//!
+//! [`ratatui-core`]: https://crates.io/crates/ratatui-core
+//! [`ratatui-widgets`]: https://crates.io/crates/ratatui-widgets
+//!
+//! # Widget Traits
+//!
+//! In Ratatui, widgets are implemented as Rust traits, which allow for easy implementation and
+//! extension. The main traits for widgets are:
+//!
+//! - [`Widget`]: Basic trait for stateless widgets that are consumed when rendered
+//! - [`StatefulWidget`]: Trait for widgets that maintain state between renders
+//! - [`WidgetRef`]: Trait for rendering widgets by reference (unstable)
+//! - [`StatefulWidgetRef`]: Trait for rendering stateful widgets by reference (unstable)
+//!
+//! ## Widget
```

Comment:
Is this also meant to be wrapped in inline code blocks like the other headers for `*Widget*`?